extensions/amp-video/0.1/video-cache.js

```diff
@@ -47,22 +47,41 @@ export function fetchCachedSources(
 
   Services.performanceFor(ampdoc.win).addEnabledExperiment('video-cache');
 
-  const {canonicalUrl, sourceUrl} = Services.documentInfoForDoc(win.document);
-  maybeReplaceSrcWithSourceElement(videoEl, win);
-  const videoUrl = resolveRelativeUrl(selectVideoSource(videoEl), sourceUrl);
-  return getCacheUrlService(videoEl, ampdoc)
-    .then((service) => service.createCacheUrl(videoUrl))
-    .then((cacheUrl) => {
-      const requestUrl = addParamsToUrl(cacheUrl.replace(/\/[ic]\//, '/mbv/'), {
-        'amp_video_host_url':
-          /* document url that contains the video */ canonicalUrl,
-        'amp_video_require_acao_header': videoEl.hasAttribute('crossorigin')
-          ? 1
-          : null,
+  let jsonResponsePromise;
+  if (shouldUseInlineVideoResponse(videoEl)) {
+    const inlineResponseEl = win.document.getElementById(
+      'amp-google-video-cache-response'
+    );
+    if (inlineResponseEl?.textContent) {
+      const inlineResponseJson = JSON.parse(inlineResponseEl.textContent);
+      jsonResponsePromise = Promise.resolve(inlineResponseJson);
+    }
+  }
+
+  if (!jsonResponsePromise) {
+    const {canonicalUrl, sourceUrl} = Services.documentInfoForDoc(win.document);
+    maybeReplaceSrcWithSourceElement(videoEl, win);
+    const videoUrl = resolveRelativeUrl(selectVideoSource(videoEl), sourceUrl);
+    jsonResponsePromise = getCacheUrlService(videoEl, ampdoc)
+      .then((service) => service.createCacheUrl(videoUrl))
+      .then((cacheUrl) => {
+        const requestUrl = addParamsToUrl(
+          cacheUrl.replace(/\/[ic]\//, '/mbv/'),
+          {
+            'amp_video_host_url':
+              /* document url that contains the video */ canonicalUrl,
+            'amp_video_require_acao_header': videoEl.hasAttribute('crossorigin')
+              ? 1
+              : null,
+          }
+        );
+        return Services.xhrFor(win)
+          .fetch(requestUrl, {prerenderSafe: true})
+          .then((xhrResponse) => xhrResponse.json());
       });
-      return Services.xhrFor(win).fetch(requestUrl, {prerenderSafe: true});
-    })
-    .then((response) => response.json())
+  }
```

Comment:
Can you extract this block that gets the config from either the script or fetches into another function?